build_tools/azure/python_nogil_lock.txt

```diff
@@ -0,0 +1,70 @@
+# The content below has been generated manually with the following command:
+# docker run -v $PWD:/scikit-learn -it nogil/python bash -c 'pip install pip-tools; pip-compile /scikit-learn/build_tools/azure/python_nogil_requirements.txt -o /scikit-learn/build_tools/azure/python_nogil_lock.txt'
+#
+# The reason behind it is that you need python-nogil to generate the pip lock
+# file. Using pip-compile --index and --extra-index will not work, for example
+# the latest cython will be picked up from PyPI, rather than the one from the
+# python-nogil index
```

Comment:
When I run the `docker run -v $PWD:/scikit-learn ...` command this header gets replaced. Maybe we move this comment into `python_nogil_requirements.txt`?

Reply:
Maybe good enough for now? Longer term, I was thinking about using Docker to generate all the pip-based lock files.

Reply:
I ended up doing what you suggested (I added the `--upgrade` option as well which is needed to upgrade dependencies as I discovered recently)